modules/codemod/lib/v7/recategorizeIconButtons.ts

```diff
@@ -106,14 +107,21 @@ export default function transformer(file: FileInfo, api: API, options: Options)
       requiredImportSpecifiers.push(buttonType);
     });
 
+  if (requiredImportSpecifiers.length === 0) {
+    return file.source;
+  }
+
   // Find all instances of IconButton within a style function
   // const StyledIconButton = styled(IconButton) gets renamed to
   // const StyledIconButton = styled(CorrectButtonMapping)
   root.find(j.VariableDeclarator).forEach(nodePath => {
     if (
       nodePath.value.init?.type === 'CallExpression' &&
       nodePath.value.init.callee.type === 'CallExpression' &&
-      nodePath.value.init.callee.arguments[0].type === 'Identifier'
+      nodePath.value.init.callee.callee.type === 'Identifier' &&
+      nodePath.value.init.callee.callee.name === 'styled' &&
+      nodePath.value.init.callee.arguments[0].type === 'Identifier' &&
+      nodePath.value.init.callee.arguments[0].name === 'IconButton'
```

Comment:
Before this change, the following would happen:

```diff
-const MyComponent = createComponent(Button)({
+const MyComponent = createComponent(TerinaryButton)({
```

Reply:
ah so you're adding more specificity?

Reply:
Yes. It is making sure the function used is named `styled` and the argument is named `IconButton`. Perhaps it should use the importmap instead in case they rename `IconButton`